quiz/quiz.go

```diff
@@ -0,0 +1,182 @@
+// This package implements the quiz, with a variation on part 2.  
+// Instead of having an timer for the entire test, it handles a timer 
+// for each question.  This was more challenging, since characters 
+// entered for previous questions had to be properly discarded
+// to get the desired behavior.
+
+package main
+
+import (
+	"bufio"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+type QuizItem struct {
+	Question string
+	Answer   string
+}
+
+type CharWithTime struct {
+	char string
+	time time.Time
+}
+
+type ErrTimeout struct {
+	message string
+}
+
+func (e *ErrTimeout) Error() string {
+	return e.message
+}
+
+func NewErrorTimeout(message string) *ErrTimeout {
+	return &ErrTimeout{message: message}
+}
+
+func OldReadQuizFromFile(filepath string) []QuizItem {
+	file, err := os.Open(filepath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	items := make([]QuizItem, 0, 100)
+	fscanner := bufio.NewScanner(file)
+	for fscanner.Scan() {
+		line := fscanner.Text()
+		// fmt.Println(line)
+		tokens := strings.Split(line, ",")
+		items = append(items, QuizItem{Question: tokens[0], Answer: tokens[1]})
+	}
+	return items
+}
+
+func ReadQuizFromFile(filepath string) []QuizItem {
+	file, err := os.Open(filepath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	items := make([]QuizItem, 0, 100)
+	r := csv.NewReader(file)
+	for {
+		record, err := r.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Println(record)
+		items = append(items, QuizItem{Question: record[0], Answer: record[1]})
+	}
+	return items
+}
+
+func ReadCharsWithTime(chars chan CharWithTime) {
```

Comment:
Why read chars instead of strings?
Also the CharWithTime struct seems very weird to me. This seems like one of the places a context would help.

Reply:
Just found a bad example googling.  Since the method is not being used I'll just delete it.